launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/TTSServlet.java

```diff
@@ -0,0 +1,245 @@
+        } else if (Strings.isNullOrEmpty(podcastLanguage)) {
+            throw new IllegalArgumentException("Unable to get Podcast Language. Please try again");
+        }
+
+        // Search for key from given feed id
+        Key desiredFeedKey = KeyFactory.stringToKey(feedKey);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity desiredFeedEntity = null;
+        try {
+            desiredFeedEntity = datastore.get(desiredFeedKey);
+        } catch (EntityNotFoundException e) {
+            e.printStackTrace();
+            res.sendError(HttpServletResponse.SC_CONFLICT, "Unable to find given URL key, Please try again");
+            return;
+        }
+
+        // Check if current user email matches feed user email
+        if (!userEmail.equals(desiredFeedEntity.getProperty(EMAIL).toString())) {
+            throw new AuthenticationException("This user does not have permission to modify this feed.");
+        }
+
+        // Turn the xml string back into an object
+        String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+        RSS rssFeed = null;
+        try {
+            rssFeed = TranslationServlet.XML_MAPPER.readValue(xmlString, RSS.class);
+        } catch (Exception e) {
+            res.sendError(HttpServletResponse.SC_CONFLICT, "Unable to translate. Try again");
+            return;
+        }
+
+        // Synthesize The podcast Text
+        ByteString synthesizedMp3 = null;
+        try {
+            synthesizedMp3 = synthesizeText(podcastText);
+        } catch (Exception e) {
+            res.sendError(HttpServletResponse.SC_CONFLICT, "unable to create mp3 from request. Please try again.");
+            return;
+        }
+        Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+        String itemCount = String.valueOf(rssFeed.getChannel().getItems().size());
+        BlobId blobId = BlobId.of(BUCKET_NAME, feedKey + itemCount);
+        BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+
+        // create an uploadble array of bytes for blobstore
+        byte[] mp3Bytes = synthesizedMp3.toByteArray();
+        storage.create(blobInfo, mp3Bytes);
+
+        // Generate mp3 link
+        String mp3Link = TTS_BASE_URL + feedKey + itemCount; // creates unique ID for each episode
+
+        rssFeed.getChannel().addItem(podcastTitle, podcastDescription, podcastLanguage, userEmail, mp3Link);
+
+        desiredFeedEntity.setProperty(XML_STRING, RSS.toXmlString(rssFeed)); // update existing entity's XML string
+
+        datastore.put(desiredFeedEntity);
+
+        Query query = new Query(LoginStatus.USER_FEED_KEY).addSort(LoginStatus.TIMESTAMP_KEY, SortDirection.DESCENDING);
+
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<UserFeed> userFeeds = new ArrayList<UserFeed>();
+        addToUserFeeds(userFeeds, results, userEmail);
```

Comment:
addToUserFeeds implies that something is being added to the user feeds, but that's not what this method does. it just converts the entities to UserFeed objects. 

change the name of the method to something like queryUserFeeds - in that method you can create the query, get the results from the database and convert the results to List<UserFeed>